lambdaclass/mirra_backend
# Add Prometheus metrics

## Motivation

## Summary of changes

- Add dependencies to have prometheus metrics and expose them in http://localhost:9568/metrics
- Add some basic in-game metrics to track tick duration, game count, and player count

## How to test it?

Setup Grafana dashboards
1. Spin up Prometheus and Grafana docker containers (`docker compose up -d`)
2. Turn on server (`make start` or whatever you use)
3. Go to http://localhost:9100 and log in, default user and password is `admin`
4. In sidebar go to `Connections > Data sources`
  4.1.  For `Prometheus server URL` set it to `http://prometheus:9090`
  4.2  Click on `Save & test`
5. In Sidebar go to `Dashboards`
  5.1. Click on `New` to show dropdown and select `Import`
  5.2 In the `Import via dashboard JSON model` section copy & paste the contents of `grafana_arena_dashboard.json` (file is in this PR) 
  5.3 Click on `Load`
  5.4 In the `prometheus` section pick your recently created data source
  5.5 Click `Import`

At this point you should be seeing some empty dashboards, now play a game (or more) and you should start to see data come into the dashboards

## Checklist
